clinica/pipelines/statistics_surface/clinica_surfstat.py

```diff
@@ -0,0 +1,151 @@
+
+    ROOT can be:
+
+        - For group comparison GLM with an interaction term:
+
+            interaction-<contrast_name>_measure-<feature_label>_fwhm-<fwhm>
+
+        - For group comparison GLM without an interaction term:
+
+            group-<group_label>_<contrast_name>_measure-<feature_label>_fwhm-<fwhm>
+
+        - For correlation GLM:
+
+            group-<group_label>_correlation-<contrast_name>-<contrast_sign>_measure-<feature_label>_fwhm-<fwhm>
+
+    Parameters
+    ----------
+    input_dir : Input folder.
+
+    output_dir : Output folder for storing results.
+
+    tsv_file : Path to the TSV file `subjects.tsv` which contains the
+        necessary metadata to run the statistical analysis.
+
+        .. warning::
+            The column names need to be accurate because they
+            are used to defined contrast and model terms.
+            Please double check for typos.
+
+    design_matrix : Design matrix in string format.
+        For example "1+Label"
+
+    contrast : The contrast to be used in the GLM.
+
+        .. warning::
+            The contrast needs to be in the design matrix.
+
+    glm_type : {"group_comparison", "correlation"}
+        Type of GLM to run:
+            - "group_comparison": Performs group comparison.
+              For example "AD - ND".
+            - "correlation": Performs correlation analysis.
+
+    group_label : Label for the group.
+        This is used in the output file names (see main description
+        of the function).
+
+    freesurfer_home : Path to the home folder of Freesurfer.
+        This is required to get the fsaverage templates.
+
+    surface_file : Path to the surface file to analyze.
+        Typically the cortical thickness.
+
+    feature_label : Label used for the measure.
+        This is used in the output file names (see main description
+        of the function).
+
+    parameters : Dictionary of additional parameters
+        - "sizeoffwhm": Smoothing. This is used in the output file names.
+          Default=20.
+        - "thresholduncorrectedpvalue": Threshold to be used with uncorrected
+          P-values. Default=0.001.
+        - "thresholdcorrectedpvalue": Threshold to be used with corrected
+          P-values. Default=0.05.
+        - "clusterthreshold": Threshold to be used to declare clusters as
+          significant. Default=0.001.
+    """
+    # Load subjects data
+    df_subjects = _read_and_check_tsv_file(tsv_file)
+    if "sizeoffwhm" in parameters:
+        fwhm = parameters["sizeoffwhm"]
+    else:
+        fwhm = DEFAULT_FWHM
+        parameters["sizeoffwhm"] = fwhm
+    surface_file = _get_t1_freesurfer_custom_file_template(input_dir)
+    thickness = _build_thickness_array(input_dir, surface_file, df_subjects, fwhm)
+
+    # Load average surface template
+    fsaverage_path = freesurfer_home / Path("subjects/fsaverage/surf")
+    average_surface, average_mesh = _get_average_surface(fsaverage_path)
```

Comment:
Apparently, BrainStat provides its own [fetchers](https://github.com/MICA-MNI/BrainStat/blob/6399b74e0076f648cc1e64fda0cd22740a5dbb78/brainstat/datasets/base.py#L110). Perhaps they could be useful?

Reply:
We can discuss this next week, I'm not really convinced that using their fetchers would be a better option compared to the solution implemented here.